lib/logstash/inputs/dead_letter_queue.rb

```diff
@@ -53,10 +58,18 @@ def register
     dlq_path = java.nio.file.Paths.get(File.join(@path, @pipeline_id))
     sincedb_path = @sincedb_path ? java.nio.file.Paths.get(@sincedb_path) : nil
     start_timestamp = @start_timestamp ? org.logstash.Timestamp.new(@start_timestamp) : nil
-    @inner_plugin = org.logstash.input.DeadLetterQueueInputPlugin.new(dlq_path, @commit_offsets, sincedb_path, start_timestamp)
+    @logstash_version = Gem::Version.new(LOGSTASH_CORE_VERSION)
+    if clean_consumed && !Gem::Requirement.new('>= 8.4.0').satisfied_by?(@logstash_version)
+      raise ConfigurationError.new("clean_consumed can be used only with Logstash version 8.4.0 and above")
+    end
+    if clean_consumed
```

Comment:
We should log or fail the pipeline if we are implicitly changing a configuration setting, particularly one that is explicitly set

Reply:
Yes we could enforce that the user sets also `commit_offsets`, I thought about the implicit setting once `clean_consumed` to smooth the adoption. But in other plugins we tend to be more explicit, so I'll switch to raising a configuration error.

Reply:
By default `commit_offset` is `true` so if the user doesn't explicitly set to `false`, the adoption is smooth.
Fixed with 114ab84f532174f87c806a46738c622a3cb2dd2e